ethexe/consensus/src/validator/mod.rs

```diff
@@ -69,25 +69,28 @@ mod subordinate;
 #[cfg(test)]
 mod tests;
 
+#[cfg(doc)]
+use self::{
+    coordinator::Coordinator, participant::Participant, producer::Producer, submitter::Submitter,
+    subordinate::Subordinate,
+};
 use crate::{
     utils::{
         BatchCommitmentValidationReply, BatchCommitmentValidationRequest,
         MultisignedBatchCommitment,
     },
+    validator::{
+        coordinator::Coordinator, participant::Participant, producer::Producer,
+        submitter::Submitter, subordinate::Subordinate,
+    },
     ConsensusEvent, ConsensusService,
 };
 use initial::Initial;
 
-#[cfg(doc)]
-use self::{
-    coordinator::Coordinator, participant::Participant, producer::Producer, submitter::Submitter,
-    subordinate::Subordinate,
-};
-
 /// The main validator service that implements the `ConsensusService` trait.
 /// This service manages the validation workflow.
 pub struct ValidatorService {
-    inner: Option<Box<dyn StateHandler>>,
+    inner: Option<ValidatorState>,
```

Comment:
Value on creation: ValidatorState::initial()
```suggestion
    state: ValidatorState,
```